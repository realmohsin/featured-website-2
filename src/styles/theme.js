import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
