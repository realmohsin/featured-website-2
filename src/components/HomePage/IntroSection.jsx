// import React from 'react'
// import { StaticImage } from 'gatsby-plugin-image'
// import { makeStyles } from '@material-ui/core/styles'
// import { Container, Grid, Typography } from '@material-ui/core'

// const useStyles = makeStyles(theme => ({
//   introSection: {
//     padding: '12rem',
//     background: `radial-gradient(circle, rgba(239,249,249, 0.9) 0%, rgba(255, 255, 255, 0.4) 100%)`,
//     position: 'relative',
//     '& .gatsby-image-wrapper-constrained': {
//       position: 'absolute',
//       top: 0,
//       bottom: 0,
//       right: 0,
//       left: 0155049
//     },
//     [theme.breakpoints.down('md')]: {
//       padding: '12rem 9rem'
//     },
//     [theme.breakpoints.down('xs')]: {
//       padding: '11rem 5rem'
//     }
//   },
//   introTitle: {
//     ...theme.custom.title2
//   },
//   introSecondaryTitle: {
//     ...theme.custom.title3
//   },
//   introGrid: {},
//   firstGridItem: {
//     paddingLeft: '4rem',
//     paddingRight: '5rem',
//     [theme.breakpoints.down('md')]: {
//       paddingLeft: 0
//     },
//     [theme.breakpoints.down('sm')]: {
//       marginBottom: '8rem'
//     },
//     [theme.breakpoints.down('xs')]: {
//       textAlign: 'center',
//       paddingRight: 0
//     }
//   },
//   secondGridItem: {
//     paddingLeft: '7rem',
//     paddingRight: '7rem',
//     '@media (max-width: 1520px)': {
//       paddingLeft: '3rem',
//       paddingRight: 0
//     },
//     [theme.breakpoints.down('md')]: {
//       paddingLeft: 0,
//       paddingRight: 0
//     }
//   },
//   paragraph: {
//     marginBottom: '2.5rem'
//   },
//   highlightedText: {
//     color: theme.palette.secondary.main,
//     letterSpacing: '0.7px',
//     fontWeight: 'normal'
//   },
//   symbolDivider: {
//     width: '3.5rem',
//     height: '2px',
//     backgroundColor: theme.palette.secondary.main,
//     marginBottom: '2rem'
//   },
//   lightBg: {
//     zIndex: -10
//   }
// }))

// const IntroSection = () => {
//   const classes = useStyles()

//   return (
//     <section className={classes.introSection}>
//       <StaticImage
//         src='../../assets/images/light-bg.png'
//         className={classes.lightBg}
//       />
//       <Container>
//         <h2 className={classes.introTitle}>Innovative Legal Counsel</h2>
//         <Grid container className={classes.introGrid}>
//           <Grid item xs={12} md={6} className={classes.firstGridItem}>
//             <h3 className={classes.introSecondaryTitle}>
//               For Businesses, Professionals & Creative People
//             </h3>
//           </Grid>
//           <Grid item xs={12} md={6} className={classes.secondGridItem}>
//             <div className={classes.symbolDivider} />
//             <Typography variant='body1' className={classes.paragraph}>
//               <strong className={classes.highlightedText}>
//                 BASED IN NEW YORK CITY
//               </strong>
//               , Romano Law is a cutting edge firm, with a passion for making a
//               difference in our clients’ lives.
//             </Typography>
//             <Typography variant='body1' className={classes.paragraph}>
//               Our legal team takes pride in developing meaningful relationships
//               with our clients. Our mission is to deliver cost-effective and
//               efficient solutions to help you or your company thrive.
//             </Typography>
//             <Typography variant='body1' className={classes.paragraph}>
//               We’re more than just attorneys – we’re trusted advisors. We love
//               what we do.
//             </Typography>
//           </Grid>
//         </Grid>
//       </Container>
//     </section>
//   )
// }

// export default IntroSection

import React from "react";
import { StaticImage } from "gatsby-plugin-image";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { Grid, Typography, Container } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  aboutSection: {
    padding: "3rem 0 12rem",
    background: `radial-gradient(circle, rgba(239,249,249, 0.9) 0%, rgba(255, 255, 255, 0.1) 100%)`,
    position: "relative",
  },
  lightBg: {
    position: "absolute !important",
    top: 0,
    bottom: 0,
    right: 0,
    left: 0,
    zIndex: -10,
    "& img": {
      objectFit: "cover",
    },
  },
  symbolDivider: {
    width: "3.5rem",
    height: "2px",
    backgroundColor: theme.palette.secondary.main,
    margin: "10rem 0 2rem",
    [theme.breakpoints.down("xs")]: {
      margin: "4rem 0 2rem",
    },
  },
  aboutTitle: {
    ...theme.custom.title2,
    textAlign: "left",
    fontSize: "5.8rem",
    paddingLeft: "5rem",
    marginBottom: "4rem",
    marginTop: "0rem",
    [theme.breakpoints.down("md")]: {
      fontSize: "4.6rem",
    },
    [theme.breakpoints.down("xs")]: {
      fontSize: "3.4rem",
      paddingLeft: "1.2rem",
      marginBottom: "3rem",
    },
  },
  aboutSubTitle: {
    ...theme.custom.title3,
    // ...theme.custom.outlineTitle,
    marginBottom: "3rem",
    [theme.breakpoints.down("md")]: {
      fontSize: "2.8rem",
    },
    [theme.breakpoints.down("xs")]: {
      fontSize: "2.1rem",
    },
  },
  gridItem: {
    padding: "0 5rem",
    [theme.breakpoints.down("xs")]: {
      padding: "0 1.2rem",
    },
  },
  contentTitle: {
    ...theme.custom.title4,
    // ...theme.custom.outlineTitle,
    letterSpacing: "1px",
    marginBottom: "3rem",
    fontSize: "1.6rem",
  },
  leftImg: {
    [theme.breakpoints.down("xs")]: {
      height: "40rem",
    },
  },
}));

const AboutSection = (props) => {
  const theme = useTheme();
  const classes = useStyles();

  const atXsDown = useMediaQuery(theme.breakpoints.down("xs")); // below 600

  const RightSideImg = () => (
    <StaticImage src="../../assets/images/about-2.jpg" alt="Business Law" />
  );

  const RightSideText = () => (
    <>
      <div className={classes.symbolDivider} data-testid="divider" />
      <h5 className={classes.contentTitle}>We Love What We Do</h5>
      <Typography variant="body1" gutterBottom>
        We are committed to providing practical legal services.
      </Typography>
      <Typography variant="body1" gutterBottom>
        Whether you’re scaling a business, negotiating a contract or trying to
        resolve a dispute – your goals become our goals.
      </Typography>
      <Typography variant="body1" gutterBottom>
        We pride ourselves in being approachable and getting things done. We
        avoid legalese. We’re proactive.
      </Typography>
      <Typography variant="body1" gutterBottom>
        Focused on business, entertainment and dispute resolution, we represent
        entrepreneurs, stake-holders and creative professionals.
      </Typography>
      <Typography variant="body1" gutterBottom>
        We focus on legal strategy, so you don’t have to.
      </Typography>
      <Typography variant="body1" gutterBottom>
        We’ve got this.
      </Typography>
    </>
  );

  return (
    <section className={classes.aboutSection}>
      <StaticImage
        src="../../assets/images/light-bg.png"
        className={classes.lightBg}
        alt="background"
      />
      <Container>
        <h1 className={classes.aboutTitle}>We are Romano Law</h1>
        <Grid container>
          <Grid item xs={12} sm={6} className={classes.gridItem}>
            <h2 className={classes.aboutSubTitle}>
              Innovative Legal Counsel for Business, Professionals & Creative
              People
            </h2>
            <StaticImage
              src="../../assets/images/about-1.jpg"
              alt="Business Law"
              className={classes.leftImg}
            />
          </Grid>
          <Grid item xs={12} sm={6} className={classes.gridItem}>
            {atXsDown ? (
              <>
                <RightSideText />
                <RightSideImg />
              </>
            ) : (
              <>
                <RightSideImg />
                <RightSideText />
              </>
            )}
          </Grid>
        </Grid>
      </Container>
    </section>
  );
};

export default AboutSection;
