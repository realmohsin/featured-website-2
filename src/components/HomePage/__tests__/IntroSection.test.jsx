import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import { createMuiTheme } from '@material-ui/core/styles';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import IntroSection from '../IntroSection';

// Mock the gatsby-plugin-image component
jest.mock('gatsby-plugin-image', () => {
  const React = require('react');
  const MockImage = (props) => {
    return <img {...props} alt={props.alt} data-testid={props['data-testid'] || 'gatsby-image'} />;
  };
  return {
    __esModule: true,
    StaticImage: MockImage,
  };
});

// Mock useMediaQuery
jest.mock('@material-ui/core/useMediaQuery');

// Mock theme
const theme = createMuiTheme({
  custom: {
    title2: {
      fontSize: '5.6rem',
      lineHeight: 1.2,
      fontWeight: 700,
      marginBottom: '2rem',
    },
    title3: {
      fontSize: '3.2rem',
      lineHeight: 1.2,
      fontWeight: 600,
      marginBottom: '1.5rem',
    },
    title4: {
      fontSize: '2.4rem',
      lineHeight: 1.3,
      fontWeight: 500,
      marginBottom: '1rem',
    },
  },
  palette: {
    secondary: {
      main: '#007bff',
    },
  },
  breakpoints: {
    down: () => '@media (max-width: 600px)',
  },
});

describe('IntroSection', () => {
  // Default to desktop view
  beforeEach(() => {
    useMediaQuery.mockReturnValue(false);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  const renderWithTheme = (component) => {
    return render(
      <ThemeProvider theme={theme}>
        {component}
      </ThemeProvider>
    );
  };

  it('renders the intro section with all main components', () => {
    renderWithTheme(<IntroSection />);
    
    // Check main title
    expect(screen.getByText('We are Romano Law')).toBeInTheDocument();
    
    // Check subtitle
    expect(screen.getByText(/Innovative Legal Counsel for Business, Professionals & Creative People/i)).toBeInTheDocument();
    
    // Check content title
    expect(screen.getByText('We Love What We Do')).toBeInTheDocument();
    
    // Check some content text
    expect(screen.getByText(/We are committed to providing practical legal services/i)).toBeInTheDocument();
    
    // Check images are rendered
    const images = screen.getAllByRole('img');
    expect(images.length).toBeGreaterThan(0);
  });

  it('displays content in correct order on desktop', () => {
    renderWithTheme(<IntroSection />);
    
    // In desktop view, the image should come before the text in the right column
    const rightColumn = screen.getAllByRole('img', { name: /Business Law/ })[1]?.parentElement;
    const divider = screen.getByTestId('divider');
    
    // Check that both elements exist
    expect(rightColumn).toBeInTheDocument();
    expect(divider).toBeInTheDocument();
    
    // Get all direct children of the right column
    const rightColumnChildren = Array.from(rightColumn?.children || []);
    
    // Find the indices of the elements we care about
    const imageIndex = rightColumnChildren.findIndex(el => 
      el === screen.getAllByRole('img', { name: /Business Law/ })[1]
    );
    const dividerIndex = rightColumnChildren.findIndex(el => 
      el === divider
    );
    
    // In desktop view, image should come before divider
    expect(imageIndex).toBeLessThan(dividerIndex);
  });

  it('displays content in correct order on mobile', () => {
    // Mock mobile view
    useMediaQuery.mockReturnValue(true);
    
    renderWithTheme(<IntroSection />);
    
    // In mobile view, the text should come before the image in the right column
    const rightColumn = screen.getAllByRole('img', { name: /Business Law/ })[1]?.parentElement;
    const divider = screen.getByTestId('divider');
    
    // Check that both elements exist
    expect(rightColumn).toBeInTheDocument();
    expect(divider).toBeInTheDocument();
    
    // Get all direct children of the right column
    const rightColumnChildren = Array.from(rightColumn?.children || []);
    
    // Find the indices of the elements we care about
    const imageIndex = rightColumnChildren.findIndex(el => 
      el === screen.getAllByRole('img', { name: /Business Law/ })[1]
    );
    const dividerIndex = rightColumnChildren.findIndex(el => 
      el === divider
    );
    
    // In mobile view, divider should come before image
    expect(dividerIndex).toBeLessThan(imageIndex);
  });

  it('matches snapshot', () => {
    const { container } = renderWithTheme(<IntroSection />);
    expect(container).toMatchSnapshot();
  });
});
