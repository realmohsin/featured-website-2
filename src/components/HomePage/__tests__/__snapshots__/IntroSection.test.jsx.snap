// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroSection matches snapshot 1`] = `
<div>
  <section
    class="makeStyles-aboutSection-25"
  >
    <img
      alt="background"
      class="makeStyles-lightBg-26"
      data-testid="gatsby-image"
      src="../../assets/images/light-bg.png"
    />
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <h1
        class="makeStyles-aboutTitle-28"
      >
        We are Romano Law
      </h1>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root makeStyles-gridItem-30 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <h2
            class="makeStyles-aboutSubTitle-29"
          >
            Innovative Legal Counsel for Business, Professionals & Creative People
          </h2>
          <img
            alt="Business Law"
            class="makeStyles-leftImg-32"
            data-testid="gatsby-image"
            src="../../assets/images/about-1.jpg"
          />
        </div>
        <div
          class="MuiGrid-root makeStyles-gridItem-30 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <img
            alt="Business Law"
            data-testid="gatsby-image"
            src="../../assets/images/about-2.jpg"
          />
          <div
            class="makeStyles-symbolDivider-27"
            data-testid="divider"
          />
          <h5
            class="makeStyles-contentTitle-31"
          >
            We Love What We Do
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            We are committed to providing practical legal services.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Whether you’re scaling a business, negotiating a contract or trying to resolve a dispute – your goals become our goals.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            We pride ourselves in being approachable and getting things done. We avoid legalese. We’re proactive.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Focused on business, entertainment and dispute resolution, we represent entrepreneurs, stake-holders and creative professionals.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            We focus on legal strategy, so you don’t have to.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            We’ve got this.
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
